src/OptionsPopup.qml

```diff
@@ -807,43 +798,61 @@ Window {
-            }
-            if (chkSetUser.checked) {
-                settings.sshUserName = fieldUserName.text
-                settings.sshUserPassword = fieldUserPassword.alreadyCrypted ? fieldUserPassword.text : imageWriter.crypt(fieldUserPassword.text)
-            }
+        var settings = { };
+        if (chkHostname.checked && fieldHostname.length) {
+            settings.hostname = fieldHostname.text
+        }
+        if (chkSetUser.checked) {
+            settings.sshUserName = fieldUserName.text
+            settings.sshUserPassword = fieldUserPassword.alreadyCrypted ? fieldUserPassword.text : imageWriter.crypt(fieldUserPassword.text)
+        }
 
-            settings.sshEnabled = chkSSH.checked
-            if (chkSSH.checked && radioPubKeyAuthentication.checked) {
-                settings.sshAuthorizedKeys = fieldPublicKey.text
-            }
-            if (chkWifi.checked) {
-                settings.wifiSSID = fieldWifiSSID.text
-                if (chkWifiSSIDHidden.checked) {
-                    settings.wifiSSIDHidden = true
-                }
-                settings.wifiPassword = fieldWifiPassword.text.length == 64 ? fieldWifiPassword.text : imageWriter.pbkdf2(fieldWifiPassword.text, fieldWifiSSID.text)
-                settings.wifiCountry = fieldWifiCountry.editText
-            }
-            if (chkLocale.checked) {
-                settings.timezone = fieldTimezone.editText
-                settings.keyboardLayout = fieldKeyboardLayout.editText
+        settings.sshEnabled = chkSSH.checked
+        if (chkSSH.checked && radioPubKeyAuthentication.checked) {
+            settings.sshAuthorizedKeys = fieldPublicKey.text
+        }
+        if (chkWifi.checked) {
+            settings.wifiSSID = fieldWifiSSID.text
+            if (chkWifiSSIDHidden.checked) {
+                settings.wifiSSIDHidden = true
             }
-
-            imageWriter.setSavedCustomizationSettings(settings)
-
-        } else if (hasSavedSettings) {
-            imageWriter.clearSavedCustomizationSettings()
-            hasSavedSettings = false
+            settings.wifiPassword = fieldWifiPassword.text.length == 64 ? fieldWifiPassword.text : imageWriter.pbkdf2(fieldWifiPassword.text, fieldWifiSSID.text)
+            settings.wifiCountry = fieldWifiCountry.editText
+        }
+        if (chkLocale.checked) {
+            settings.timezone = fieldTimezone.editText
+            settings.keyboardLayout = fieldKeyboardLayout.editText
         }
 
         imageWriter.setSetting("beep", chkBeep.checked)
         imageWriter.setSetting("eject", chkEject.checked)
         imageWriter.setSetting("telemetry", chkTelemtry.checked)
+
+        if (chkHostname.checked || chkSetUser.checked || chkSSH.checked || chkWifi.checked || chkLocale.checked) {
+            /* OS customization to be applied. */
+            hasSavedSettings = true
+            saveSettingsSignal(settings)
+        }
+    }
+
+    function clearCustomizationFields()
```

Comment:
This function has an unintended side-effect.  Normally when you open "Edit settings" with no customisation saved many fields are prefilled with something helpful, e.g. the hostname is prefilled with "raspberrypi" and the username is prefilled with your host OS username.  But if you clear settings, then cancel imaging at the warning prompt, then go back into edit settings, then all the fields are totally blank and don't have the helpful prefills.

Reply:
Addressed in [819ffe1](https://github.com/raspberrypi/rpi-imager/pull/677/commits/819ffe14f0e84bd08b60362b672db3a1dd632f93)